### PR TITLE
[types/fbt] Add types for `fbs`

### PR DESCRIPTION
**Pull request by taozhou-glean**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [fbs](https://facebook.github.io/fbt/docs/enforcing_plain_text), [1.0.0 changelog](https://facebook.github.io/fbt/blog/2022/04/12/blog-post/#whats-changed)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/fbt/index.d.ts [hunk 1]
```diff
@@ -222,7 +217,19 @@ interface FbtHookRegistrations {
 
 export function init(options: { hooks?: FbtHookRegistrations; translations: TranslationDict }): void;
 
-export const fbt: Fbt;
+interface FbtOutput {
+    fbt: string[];
```
Comment (taozhou-glean) [inline types/fbt/index.d.ts:221]: I am not sure why this was `string[]` before, should not it be `string` or maybe `ReactNode/ReactElement` instead ?

Comment (retyui) [reply types/fbt/index.d.ts:221]:
```
<fbt desc="param example">
  Hello,
  <fbt:param name="name">{person.getName()}</fbt:param>.
</fbt>
// ^^^ the result will be an array
```

**Comment by retyui** [comment]
@taozhou-glean Update the readme file:  `Provide a URL to documentation or source code which provides context for the suggested changes`

File: types/fbt/index.d.ts (hunk 1 above)
Comment (taozhou-glean) [reply types/fbt/index.d.ts:221]: but it's using it as a JSX element, not function call right ? so at least for `fb()`, should we change it to return string instead ? right now, a lot warnings show up when try to assign `fbt()` just to strings

**Comment by taozhou-glean** [comment]
> @taozhou-glean Update the readme file: `Provide a URL to documentation or source code which provides context for the suggested changes`

Done!

File: types/fbt/index.d.ts (hunk 1 above)
Comment (retyui) [reply types/fbt/index.d.ts:221]: I did a couple tests, and this how the fbt result represented  in JS runtime:

```tsx
import { isValidElement } from 'react';
import {GenderConst, IntlVariations, fbs, fbt, init} from 'fbt';

const withParams = fbt('Hello, ' + fbt.param('name', 'retyui'), 'param example');

typeof withParams; // object
withParams.toString() // "Hello, retyui"
JSON.stringify(withParams) // "\"Hello, retyui\""
isValidElement(withParams) // true
```

@taozhou-glean What do you think will be the best return type for `fbt()`  ?

**Review by retyui — APPROVED** [review]
Very well done!

File: types/fbt/index.d.ts (hunk 1 above)
Comment (taozhou-glean) [reply types/fbt/index.d.ts:221]: I think `string` would be very convenient given that's probably the expected type in most cases. Tho with `fbs`, I think `ReactNode` might be more accurate, and it's also recommended to use `fbs` for simple strings and `fbt` when html included.

Comment (davydspotnana) [reply types/fbt/index.d.ts:221]: ok, let's replace `string[]` with `string`

**Comment by taozhou-glean** [comment]
Ready to merge

**Comment by retyui** [comment]
@taozhou-glean why you merged PR without replacing  `string[]` with `string` ?

**Comment by taozhou-glean** [comment]
> @taozhou-glean why you merged PR without replacing `string[]` with `string` ?

I think we have not yet agreed whether we should use `string[]` or `string` or `ReactNode` yet, so I decide to just keep it as is and we can follow up with a separate PR once it's decided ;)

**Pull request merged**